### PR TITLE
[BEAM-1830] KafkaIO : Add withTopic() api that takes single topic.

### PR DESCRIPTION
**Pull request by rangadi**
Overwhelming majority of KafkaIO readers consume just one topic. It would be nice to have `withTopic(topic)` rather than always requiring a list. 

In addition, I am fixing a small bug I noticed while using `KafkaIO.write().values()`.  Remove need for setting key coder for Writer while writing values only. If we didn't specifiy the key coder, validation succeeded but it failed a check while instantiating Kafka producer.

### CODE REVIEW

**Comment by rangadi** [comment]
+R: @amitsela +R: @jkff

File: sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java [hunk 1]
```diff
@@ -1159,7 +1167,6 @@ public void close() throws IOException {
     @Nullable abstract String getTopic();
     @Nullable abstract Coder<K> getKeyCoder();
     @Nullable abstract Coder<V> getValueCoder();
-    abstract boolean getValueOnly();
```
Comment (jkff) [inline sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java]: This and the changes below don't seem to be related to the current commit or PR. Also, all of these changes are missing tests.

Comment (rangadi) [inline sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java]: yeah, mentioned about it in the description. This is a bug I noticed when I tried to use the writer. 

`withTopic()` is just a wrapper/builder method. All the existing tests use multiple topics, should I another test with single topic?

Testing sink changes are partially done: The validation code is tested in `KafkaIOTest.testValuesSink()`. This test didn't fail earlier  since we don't use a real KafkaProducer that instantiates the serializers. Let me see if I can manually try to instantiate.

Comment (jkff) [inline sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java]: Either way please split into multiple commits (ideally multiple PRs, since these commits are not logically related)

Comment (rangadi) [inline sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java]: Reverted writer fix. Added a test with single topic that uses new API.

**Comment by rangadi** [comment]
Thanks Eugene. PTAL. 
Reverted writer fix. Added a test with single topic that uses new API.

**Comment by rangadi** [comment]
Kafka writer is fixed in #2369.

File: sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/KafkaIOTest.java [hunk 2]
```diff
@@ -293,6 +293,31 @@ public void testUnboundedSource() {
 
   @Test
   @Category(NeedsRunner.class)
```
Comment (jkff) [inline sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/KafkaIOTest.java]: Per https://github.com/apache/beam/pull/2359 we no longer need NeedsRunner.

**Comment by amitsela** [comment]
+R: @amitsela as Eugene seems to have this.

File: sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/KafkaIOTest.java (hunk 2 above)
Comment (rangadi) [inline sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/KafkaIOTest.java]: Nice. Removed it for the new test.

**Comment by rangadi** [comment]
@jkff Thanks for the review. It might be better for you to squash while merging. That way PR retains the commit history.

**Comment by rangadi** [comment]
retest this please.

**Comment by rangadi** [comment]
retest this please.
